case-lib/lib.sh

```diff
@@ -1187,3 +1187,39 @@ perf_analyze()
     fi
 }
 
+# Restore the machine state from a file.
+# Should be triggered at the end of
+# every test touching ALSA.
+# Couple with save_machine_state.
+# Passes through the status of the previous command
+# so as to not interefere with the func_exit_handler
+# param1: file name
+restore_alsa_state()
+{
+    status=$?
+    dlogi "restore_alsa_state called with ${1}"
+    if [ -f "$1" ]; then
+        dlogi "restore_alsa_state found relevant state file."
+        alsactl restore --file /var/tmp/"$1" --pedantic --no-ucm --no-init-fallback || dlogi "alsactl state restoration failed!"
+        rm /var/tmp/"$1" || dlogi "Old state file removal failed!"
+    fi
+    return "$status"
```

Comment:
I'm not a fan of this... for instance it might print a misleading stack trace if the previous command failed.